packages/block-library/src/comments-query-loop/edit.js

```diff
@@ -2,51 +2,47 @@
  * WordPress dependencies
  */
 import {
-	BlockControls,
-	InspectorControls,
 	useBlockProps,
 	useInnerBlocksProps,
+	store as blockEditorStore,
 } from '@wordpress/block-editor';
-import { SelectControl } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import QueryToolbar from './toolbar';
+import CommentsInspectorControls from './edit/comments-inspector-controls';
+import { useSelect } from '@wordpress/data';
 
 const TEMPLATE = [
 	[ 'core/comment-template' ],
 	[ 'core/comments-pagination' ],
 ];
 
 export default function CommentsQueryLoopEdit( { attributes, setAttributes } ) {
-	const { perPage, tagName: TagName } = attributes;
+	const { tagName: TagName } = attributes;
 
 	const blockProps = useBlockProps();
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		template: TEMPLATE,
 	} );
 
+	const {
+		__experimentalDiscussionSettings: { commentOrder, commentsPerPage },
+	} = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		return getSettings();
```

Comment:
I recommend returning only `__experimentalDiscussionSettings` here, because otherwise, this component will re-render every time any setting changes.